### PR TITLE
[alsa] Better handle plugin directory

### LINKED ISSUE #26161: [alsa] Search for alsa plugins in system directory
****

**Is your feature request related to a problem? Please describe.**

ALSA plugins reside in a local directory in the same path as libasound.so file, called `alsa-lib`. Those plugins get loaded by libasound.so at runtime.

However, libasound.so doesn't look for the plugins inside the system lib directory, but in the current one. This means that unless you're using pure ALSA, your binary will fail to start (since vcpkg's libasound.so will try to look for Pipewire plugins inside the vcpkg lib directory where they're not).

For example, on Fedora (which switched completely to Pipewire), an application using the [alsa] port willl always fail to start because libasound.so can't find the pipewire ALSA plugin.

**Proposed solution**

[alsa] should be patched to look for plugins inside the system directory (/usr/lib). However, I do not have enough knowledge to say whether this is the correct, since the plugins are more likely than not compiled for a different version of ALSA than the one in vcpkg.

**Describe alternatives you've considered**

Provide the alsa plugins using vcpkg as well. I'm not sure how feasible this is however.

A fix for this would be highly appreciated, as this makes the [alsa] port completely unusable on Fedora (as well as on other distros which use Pipewire).

### PR DESCRIPTION
**Pull request by nfarid**
**Describe the pull request**

- #### What does your PR fix?
  Fixes #26161, but also handles debian based distros as well.  

Alsa looks for it's plugins in its installed directory. Typically alsa is typically installed via a system package manager, so this works.
However when it's installed with vcpkg, it hardcodes the vcpkg directory as its plugin directory, which makes it broken, as stated in the issue above.
What this PR does it allow ALSA to also search for the system directory.  
Debian based distro have the alsa plugin directory as /usr/lib/[arch]/alsa-lib, while other distros would have the alsa plugin directory /usr/lib[64]/alsa-lib, so this needs to be considered.

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  x64-linux, No

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  Yes

### CODE REVIEW

**Review by LilyWangLL — DISMISSED** [review]

File: ports/alsa/fix-plugin-dir.patch [hunk 1]
```diff
@@ -1,13 +1,47 @@
...
++#include <dirent.h>
++
+ #if defined(HAVE_LIBDL) && defined(__GLIBC__) && !defined(__UCLIBC__)
+ #define DL_ORIGIN_AVAILABLE 1
+ #endif
+@@ -79,6 +82,16 @@ static void snd_dlinfo_origin(char *path, size_t path_len)
+ #endif
+ }
+ 
++static unsigned int checkExist(const char* dirName) {
```
Comment (dan-shaw) [inline ports/alsa/fix-plugin-dir.patch:19]: Can these changes be submitted upstream?

Comment (nfarid) [reply ports/alsa/fix-plugin-dir.patch:19]: I will, once this I'm finish with this PR.

**Comment by nfarid** [comment]
Shouldn't this be in category:port-bug instead of category:port-feature? Since it's just fixing the issue above.

**Comment by vicroms** [comment]
Need to test that this works correctly with the changes introduced in https://github.com/microsoft/vcpkg-tool/pull/814

**Pull request merged**